kyuubi-main/src/test/scala/org/apache/kyuubi/operation/WithKyuubiServer.scala

```diff
@@ -19,22 +19,33 @@ package org.apache.kyuubi.operation
 
 import java.nio.file.Files
 
-import org.apache.kyuubi.{KyuubiFunSuite, Utils}
+import org.apache.kyuubi.Utils
 import org.apache.kyuubi.config.KyuubiConf
 import org.apache.kyuubi.config.KyuubiConf._
+import org.apache.kyuubi.config.internal.Tests.TESTING_YARN_CONF_DIR
 import org.apache.kyuubi.ha.HighAvailabilityConf.{HA_ZK_ACL_ENABLED, HA_ZK_QUORUM}
 import org.apache.kyuubi.server.KyuubiServer
 import org.apache.kyuubi.zookeeper.{EmbeddedZookeeper, ZookeeperConf}
 
-trait WithKyuubiServer extends KyuubiFunSuite {
+trait WithKyuubiServer extends WithMiniYarnCluster {
 
   protected val conf: KyuubiConf
 
   private var zkServer: EmbeddedZookeeper = _
   private var server: KyuubiServer = _
   private val metastore = Utils.createTempDir()
 
+  override protected val startMiniYarnCluster: Boolean = false
+
   override def beforeAll(): Unit = {
+    super.beforeAll()
+    if (startMiniYarnCluster) {
+      conf.set("spark.master", "yarn")
+      conf.set("spark.yarn.queue", "root.default")
+      conf.set("spark.executors.instances", "1")
```

Comment:
Does these configs really used ? Kyuubi config cann't pass to Spark as far as I know.

Reply:
spark.master need to be set, it could be transferred to spark sql engine.

Reply:
ah you are right, the related code is in `SparkProcessBuilder`. 

The config `conf.set("spark.master", "yarn")` seems necessary, but the others config should use `setIfMissing` instead of `set` ?